testing/test_assertrewrite.py

```diff
@@ -1197,6 +1197,27 @@ def test_simple_failure():
         result.stdout.fnmatch_lines("*E*assert (1 + 1) == 3")
 
 
+def test_source_mtime_long_long(testdir):
+    """Support modification dates after 2038 in rewritten files (#4903).
+
+    pytest would crash with:
+
+            fp.write(struct.pack("<ll", mtime, size))
+        E   struct.error: argument out of range
+    """
+    p = testdir.makepyfile(
+        """
+        def test(): pass
+    """
+    )
+    # use unsigned long timestamp which overflows signed long,
+    # which was the cause of the bug
+    timestamp = 2 ** 32 - 1
```

Comment:
You could add another test with `2**32 + 1` - that is where the added 0xFFFFFFFF of my patch comes into play.

Reply:
Good suggestion, done. 👍